### PR TITLE
[MRG] Pip Installation

### LINKED ISSUE #139: Pip Installation
#### Software and Hardware Versions 
N.A.

#### Problem description
Require of pip installation of modelci:
```
pip install modelci
```

#### Steps to Reproduce the Problem
N.A.

#### Expected Behavior
N.A.

#### Other Information 
N.A.

### PR DESCRIPTION
**Pull request by YuanmingLeee**
Closes #139 

#### What has been done to verify that this works as intended?
```shell script
pip install .
```
The output shall be:
```
Successfully installed modelci-1.0.0
```

#### Why is this the best possible solution? Were any other approaches considered?
N.A.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
User now can use `pip` to set up their environment.

#### Does this change require updates to documentation?
Yes, updated.

#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+with open('requirements.txt') as f:
+    install_requires = list()
+    for line in f.readlines():
+        install_requires.append(line.strip())
+
+setup(
+    name='modelci',
+    version='1.0.0',
+    description='A complete platform for managing, converting, profiling, and deploying models as cloud services (MLaaS)',
+    author='Yuanming Li',
```
Comment (HuaizhengZhang) [inline setup.py:93]: Author changed to "NTU CAP"
author email ‘huaizhen001@e.ntu.edu.sg

File: modelci/hub/__init__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+Email: yli056@e.ntu.edu.sg
+Date: 9/19/2020
+"""
+__all__ = ['converter', 'manager', 'client', 'utils', 'Profiler']
+
+from . import client
+from . import converter
+from . import manager
+from . import utils
+from .profiler import Profiler
```
Comment (HuaizhengZhang) [inline modelci/hub/__init__.py:14]: as profiler?

all the others are lowercase

**Review by HuaizhengZhang — DISMISSED** [review]

File: modelci/hub/__init__.py (hunk 2 above)
Comment (YuanmingLeee) [reply modelci/hub/__init__.py:14]: because `Profiler` is a class, others are modules.

File: setup.py (hunk 1 above)
Comment (YuanmingLeee) [reply setup.py:93]: refactored

**Comment by huangyz0918** [comment]
should not be `pip install .`, should be something like `pip install --upgrade git+url`.

File: README.md [hunk 3]
```diff
@@ -40,6 +40,12 @@
 
 ## Installation
 
+### Pip
```
Comment (huangyz0918) [inline README.md:43]: pip is a kind of command line I think...

File: modelci/types/proto/service_pb2.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+_INFERREQUEST = _descriptor.Descriptor(
+    name='InferRequest',
+    full_name='InferRequest',
+    filename=None,
+    file=DESCRIPTOR,
+    containing_type=None,
+    fields=[
+        _descriptor.FieldDescriptor(
+            name='model_name', full_name='InferRequest.model_name', index=0,
+            number=1, type=9, cpp_type=9, label=1,
```
Comment (huangyz0918) [inline modelci/types/proto/service_pb2.py:34]: Are those two grpcio generated files related to the type of OS?

**Comment by huangyz0918** [comment]
A suggestion is you should refactor the code and fix the existing issues first, the installation should be the last task to do, otherwise you may need to fix the installation things if you have deleted any packages or change the names, etc. @YuanmingLeee This PR will be merged after those work done.

File: modelci/types/proto/service_pb2.py (hunk 4 above)
Comment (YuanmingLeee) [reply modelci/types/proto/service_pb2.py:34]: I tested on Windows, with the same version of grpcio-tools. It has slight differences. I think those difference are not important:
`service_pb2.py`
image
A new kwargs provided. Missing of this args only create a warning.

`service_pb2_grpc.py`
image
A new class related to external API. Not sure what will this do, we do not use this class anywhere.

File: README.md (hunk 3 above)
Comment (YuanmingLeee) [reply README.md]: refactored to `using pip`

**Comment by huangyz0918** [comment]
please remove the relative imports to fix lints.

**Comment by HuaizhengZhang** [comment]
conflict should be resovled.

**Review by HuaizhengZhang — APPROVED** [review]

**Pull request merged**